### PR TITLE
add context to logging of supervisor work queue interaction

### PR DESCRIPTION
**Pull request by bmc-msft**

### CODE REVIEW

**Comment by bmc-msft** [comment]
This includes the changes from #602 to enable builds

File: src/agent/onefuzz-supervisor/src/work.rs [hunk 1]
```diff
@@ -159,25 +162,25 @@ impl WorkQueue {
         // it was just due to a stale SAS URL.
         if let Err(err) = &msg {
             if is_auth_error(err) {
-                self.renew().await?;
+                self.renew().await.context("unable to renew registration")?;
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/work.rs:165]: Distinguish this context (renewal attempt due to auth error) from the above.

File: src/agent/onefuzz-supervisor/src/work.rs [hunk 2]
```diff
@@ -189,8 +192,11 @@ impl WorkQueue {
         // it was just due to a stale SAS URL.
         if let Err(err) = &result {
             if is_auth_error(err) {
-                self.renew().await?;
-                self.queue.delete(receipt).await?;
+                self.renew().await.context("unable to renew registration")?;
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/work.rs:195]: Distinguish this context (renewal due to auth error while claiming work) from others.

File: src/agent/storage-queue/src/lib.rs [hunk 3]
```diff
@@ -57,15 +60,23 @@ impl QueueClient {
             .http
             .get(self.messages_url())
             .send_retry_default()
-            .await?
-            .error_for_status()?;
-        let text = response.text().await?;
+            .await
+            .context("storage queue delete failed")?
```
Comment (ranweiler) [inline src/agent/storage-queue/src/lib.rs:64]:
```suggestion
            .context("storage queue pop failed")?
```

File: src/agent/storage-queue/src/lib.rs [hunk 4]
```diff
@@ -145,11 +158,17 @@ impl Message {
...
     pub fn get<'a, T: serde::de::Deserialize<'a>>(&'a self) -> Result<T> {
-        let data = serde_json::from_slice(&self.data)?;
+        let data =
+            serde_json::from_slice(&self.data).context("get storage queue message failed")?;
         Ok(data)
     }
 }
 
+fn is_empty_message(text: &str) -> bool {
+    text.contains(r"<QueueMessagesList></QueueMessagesList>")
```
Comment (ranweiler) [inline src/agent/storage-queue/src/lib.rs:168]: Would be good to add a comment with this link: `https://docs.microsoft.com/en-us/rest/api/storageservices/get-messages#response-body`

This check is a bit brittle, it'd be nice if we could handle these cases:
```
<QueueMessagesList>  </QueueMessagesList>
```
```
<QueueMessagesList>
</QueueMessagesList>
```

**Review by ranweiler — APPROVED** [review]

**Review by ranweiler — APPROVED** [review]

**Pull request merged**